django_select2/admin.py

```diff
@@ -0,0 +1,133 @@
+
+from .forms import (
+    ModelSelect2Mixin, ModelSelect2MultipleWidget, ModelSelect2Widget,
+    Select2Widget
+)
+
+
+class Select2ModelAdminMixin(object):
+    """
+    Base mixin that allows to define and set select2 widgets on model fields.
+
+    It relies on some Django admin.options.BaseModelAdmin attributes and needs
+    a single configuration parameter to activate select2 on specific fields.
+
+    ``select_2_fields`` is a dictionary of model field names mapped to a
+    configuration dictionary which might be empty, to rely on a fully automatic
+    configuration, or containing one or all of the optional settings:
+        - ``widget``: user defined widget for that field (e.g. Select2Widget),
+          when not provided, the widget is assigned automatically;
+        - ``widget_kwargs``: custom arguments for widget initialization
+          (e.g. {'search_fields': ['name__icontains']}).
+
+    """
+
+    select2_fields = {}
+
+    def get_widgets(self):
+        """Assign select2 widgets as defined via the select2_fields dict."""
+        widgets = {}
+        for field_name, field_options in self.select2_fields.items():
+            user_widget = field_options.get('widget')
+            widget_kwargs = field_options.get('widget_kwargs', {})
+            db_field = self.model._meta.get_field(field_name)
+            field_is_m2m = isinstance(db_field, ManyToManyField)
+            field_is_fk = isinstance(db_field, ForeignKey)
+            if field_is_fk or field_is_m2m:
+                widget = (
+                    user_widget or field_is_fk and ModelSelect2Widget or
+                    ModelSelect2MultipleWidget
+                )
+                if issubclass(widget, ModelSelect2Mixin):
+                    related_model = (
+                        hasattr(db_field, 'remote_field') and
+                        db_field.remote_field or db_field.rel
+                    ).model
+                    # FIXME Change previous line when Django 1.8 is unsupported
+                    widget_kwargs['model'] = related_model
+                    widget_kwargs.setdefault('attrs', {}).update({
+                        'data-related-model': '{}.{}'.format(
+                            related_model._meta.app_label,
+                            related_model._meta.model_name,
+                        ),
+                    })
+            else:
+                widget = user_widget or Select2Widget
+                if db_field.choices:
+                    widget_kwargs['choices'] = db_field.get_choices(
+                        include_blank=db_field.blank,
+                        blank_choice=[('', 'None')]
+                    )
+            widgets[field_name] = widget(**widget_kwargs)
+        return widgets
+
+    class Media:
+        """Assure the correct js scripts import sequence."""
+
+        js = (
+            '//ajax.googleapis.com/ajax/libs/jquery/2.2.4/jquery.min.js',
```

Comment:
Yes, importing that external library that way is simply wrong, sorry. I actually saw it coming, sorry again. That said, in the discussion related to your Django integration proposal, I have seen that the scripts import problem is still quite an issue for everyone trying to integrate select2, particularly in Django admin. Therefore, once we remove that script from there, how would you prefer to deal with the situation? There stays the need to provide select2 with a jQuery library, besides it must be imported strictly before the other scripts. Personally, I would like the admin to be as much _out-of-the-box_ as possible, therefore I would prefer not having to force the user to always add its own jQuery library manually. I am sure you know this problem very well already and probably already solved it someway. Should I add a custom setting for it? I am totally open to discussion on this point.